### PR TITLE
use headers then fallback to RemoteAddr when getting remote IP

### LINKED ISSUE #63: use X-Real-IP / X-Forwarded-For instead of remoteAddr
@Dyrk and @chappjc are reporting that remoteAddr always gives 127.0.0.1 due to nginx proxying.

It seems to work fine for me but it may be due to how to the environment I'm running it in.

Anyway, pretty simple to fix.

### PR DESCRIPTION
**Pull request by jolan**
Lightly tested and seems to work.

I also dos2unix'd the nginx.conf and changed X-Real-IP to X-Real-Ip since it was converting to that anyway due to the mime header specification.

I figured out why it worked for me -- I was running it on my firewall/nat box and it was spitting out my external address instead of 127.0.0.1 every time so it looked like it was working.  I tested with lynx from a VPS this time.

Fixes #63.

### CODE REVIEW

**Comment by jolan** [comment]
@dyrk @chappjc If you could review/test/confirm that'd be great.

File: controllers/main.go [hunk 1]
```diff
@@ -28,6 +30,40 @@ import (
 	"gopkg.in/gorp.v1"
 )
 
+// ipRange is a structure that holds the start and end of a range of ip addresses
+type ipRange struct {
```
Comment (jrick) [inline controllers/main.go]: Wouldn't it be more appropriate/standard to use net.IPNet instead?  Then checking whether an IP is in a network is just `ipNet.Contains(ip)`.

**Comment by chappjc** [comment]
This certainly might work, but for a reverse proxy through localhost, it seems like you can configure nginx (>=1.9) with the standard real_ip module like:

```
set_real_ip_from 127.0.0.1;
real_ip_header X-Forwarded-For;
real_ip_recursive on;
```

Then:

    proxy_set_header X-Real-IP $realip_remote_addr;

instead of

    proxy_set_header X-Real-IP $remote_addr;

nginx should work the magic for us.

**Comment by jolan** [comment]
Yeah I'd prefer to just have nginx handle it but not totally sure I understand how to correctly set it up either.

**Comment by chappjc** [comment]
Ah, my mistake was thinking http.Request.RealAddr comes from X-Real-IP.  This actually works:

https://github.com/chappjc/dcrstakepool/tree/realip

I can squash and PR.

**Comment by chappjc** [comment]
Just another option:
https://github.com/decred/dcrstakepool/pull/66

**Comment by jolan** [comment]
Closing in favor of #66.

**Pull request closed without merging**